code/modules/admin/randomverbs.dm

```diff
@@ -2968,3 +2968,60 @@ var/global/force_radio_maptext = FALSE
 				return
 	else
 		boutput(src, "You must be at least an Administrator to use this command.")
+/client/proc/toggle_all_artifacts()
+	// Housekeeping
+	SET_ADMIN_CAT(ADMIN_CAT_FUN)
+	set name = "Toggle All Artifacts"
+	if (holder && src.holder.level >= LEVEL_ADMIN)
+		switch(alert("What would you like to do?",,"Activate All","Deactivate All","Cancel"))
+			if("Cancel")
+				return
+			if("Activate All")
+				// You definitely can activate an artnuke like this maybe be sure we're sure
+				switch(alert("ARE YOU SURE????","HEY WOAH THERE","Yes","No"))
+					if(!"Yes")
+						return
+				logTheThing(LOG_ADMIN, src, "started activating all available artifacts.")
+				message_admins("[key_name(usr)] started activating all available artifacts.")
+				for(var/artifact as anything in by_cat[TR_CAT_ARTIFACTS])
+					var/obj/holder = artifact
+					holder.ArtifactActivated()
+			if("Deactivate All")
+				// No confirmation for this since you cant really activate an artnuke like this
+				logTheThing(LOG_ADMIN, src, "started deactivating all available artifacts.")
+				message_admins("[key_name(usr)] started deactivating all available artifacts.")
+				for(var/artifact as anything in by_cat[TR_CAT_ARTIFACTS])
+					var/obj/holder = artifact
+					holder.ArtifactDeactivated()
+	else
+		boutput(src, "You must be at least an Administrator to use this command.")
+
+/client/proc/spawn_tons_of_artifacts()
+	var/artifact_bulk_limit = 100 // only this many artifacts per second
+	var/artifact_hard_limit = 2000 // Stops you from putting in 99999999999 and killing the server
+	// Housekeeping
+	SET_ADMIN_CAT(ADMIN_CAT_FUN)
+	set name = "Bulk Spawn Artifacts"
+	if (holder && src.holder.level >= LEVEL_ADMIN)
+		switch(alert("THIS COULD SERIOUSLY FUCK THINGS UP. That being said, do it anyway?","WOAH THERE BUCKAROO","Yes","No"))
+			if("No")
+				return
+			if("Yes")
+				// The actual stuff
+				var/artifacts_spawned = 0
+				var/amt_artifacts = input(usr,"How many artifacts do we want to spawn?\n(Default: 50)",,50) as num
+				amt_artifacts = min(amt_artifacts, artifact_hard_limit)
+				var/floors_only = (alert(src, "Only spawn artifacts on floors?",, "Yes", "No") == "Yes")
+				logTheThing(LOG_ADMIN, src, "started spawning [amt_artifacts] artifacts.")
+				message_admins("[key_name(usr)] started spawning [amt_artifacts] artifacts.")
+				while (artifacts_spawned < amt_artifacts)
+					var/turf/T = locate(rand(1, world.maxx), rand(1, world.maxy), Z_LEVEL_STATION)
+					if (floors_only && !istype(T, /turf/simulated/floor))
+						continue
```

Comment:
would still spawn stuff inside windows/doors/etc.
not necessarily an issue, just something to consider

Reply:
yeah thats fair and something observed in the clip, its worth changing if its going to see use. fwiw radiation blowouts also do that (and operate off a 0.5% chance per tile to spawn an artifact, going through every turf on station ever) and might be worth changing as well? though it mostly works as is and the probability based approach scales amounts of artifacts to map size which is cool

yeah fixing that though

Reply:
pushed commit to resolve